apps/src/fish/Fish.js

```diff
@@ -0,0 +1,124 @@
+  this.studioApp_ = null;
+};
+
+/**
+ * Inject the studioApp singleton.
+ */
+Fish.prototype.injectStudioApp = function(studioApp) {
+  this.studioApp_ = studioApp;
+};
+
+/**
+ * Initialize this WebLab instance.  Called on page load.
+ */
+Fish.prototype.init = function(config) {
+  if (!this.studioApp_) {
+    throw new Error('Fish requires a StudioApp');
+  }
+
+  this.skin = config.skin;
+  this.level = config.level;
+
+  config.makeYourOwn = false;
+  config.wireframeShare = true;
+  config.noHowItWorks = true;
+
+  config.pinWorkspaceToBottom = true;
+
+  const onMount = () => {
+    // NOTE: Other apps call studioApp.init(), except WebLab. Fish is imitating WebLab
+    this.studioApp_.setConfigValues_(config);
+
+    // NOTE: if we called studioApp_.init(), the code here would be executed
+    // automatically since pinWorkspaceToBottom is true...
+    const container = document.getElementById(config.containerId);
+    const bodyElement = document.body;
+    bodyElement.style.overflow = 'hidden';
+    bodyElement.className = bodyElement.className + ' pin_bottom';
+    container.className = container.className + ' pin_bottom';
+
+    this.initMLActivities();
+  };
+
+  // Push initial level properties into the Redux store
+  this.studioApp_.setPageConstants(config, {
+    channelId: config.channel,
+    noVisualization: true,
+    visualizationInWorkspace: true,
+    isProjectLevel: !!config.level.isProjectLevel
+  });
+
+  ReactDOM.render(
+    <Provider store={getStore()}>
+      <FishView onMount={onMount} />
+    </Provider>,
+    document.getElementById(config.containerId)
+  );
+};
+
+Fish.prototype.initMLActivities = function() {
+  // Set up initial state
+  const canvas = document.getElementById('activity-canvas');
+  const backgroundCanvas = document.getElementById('background-canvas');
+  canvas.width = backgroundCanvas.width = constants.canvasWidth;
+  canvas.height = backgroundCanvas.height = constants.canvasHeight;
+
+  // Temporarily use URL parameter to set some state.
+  const smallWordSet = window.location.href.indexOf('words=small') !== -1;
+
+  // Set initial state for UI elements.
+  const state = setState({
+    currentMode: Modes.Loading,
+    canvas,
+    backgroundCanvas,
+    uiContainer: document.getElementById('ui-container'),
+    headerContainer: document.getElementById('header-container'),
+    footerContainer: document.getElementById('footer-container'),
+    smallWordSet
+  });
+
+  // Initialize our first model.
+  initModel(state);
+
+  // Start the canvas renderer.  It will self-perpetute by calling
+  // requestAnimationFrame on itself.
+  renderCanvas();
+
+  // Render the UI.
+  renderUI();
+
+  // And have the render UI handler be called every time state is set.
+  setSetStateCallback(renderUI);
+};
+
+// Tell React to explicitly render the UI.
+export const renderUI = () => {
+  const renderElement = document.getElementById('container-react');
+  ReactDOM.render(<UI />, renderElement);
+};
```

Comment:
As discussed, it does feel that some of this would be common to both source trees and could remain common code rather than being duplicated.